# FY21 United Way of Southeast Louisiana (UWSELA) Interim Report Data Request

For ticket #12565 submitted January 21, 2022.

Prepared by Baker Renneckar

In [1]:
%load_ext autoreload
%autoreload 2

In [3]:
import pandas as pd
from datetime import datetime
import numpy as np
# from tabulate import tabulate
# import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')


In [5]:
# %matplotlib inline

In [6]:
df = pd.read_pickle('../data/processed/merged_data_ay_2020_21.pkl')

## Demographic Data



#### Gender / Age / Student Type Breakdown

In [7]:
pd.crosstab([df.C_Gender__c, df.C_Age__c], df.RT_RecordType__c, margins=True, colnames=[
            'Student Type'], rownames=['Gender', 'Age'], margins_name='Total')

Student Type  College Student  High School Student  Total
Gender Age                                               
Female 15.0                 0                   22     22
       16.0                 0                   27     27
       17.0                 0                   45     45
       18.0                24                   13     37
       19.0                30                    0     30
       20.0                31                    0     31
       21.0                29                    0     29
       22.0                17                    0     17
       23.0                12                    0     12
       24.0                 5                    0      5
       25.0                 2                    0      2
       26.0                 1                    0      1
       27.0                 1                    0      1
Male   15.0                 0                    6      6
       16.0                 0                    9      9
       17.0                 0                   21     21
       18.0                12                    4     16
       19.0                16                    0     16
       20.0                 8                    0      8
       21.0                10                    0     10
       22.0                 6                    0      6
       23.0                 2                    0      2
       24.0                 1                    0      1
       25.0                 1                    0      1
Other  18.0                 0                    1      1
Total                     208                  148    356

#### Gender / Ethnic / Student Type Breakdown

In [8]:
pd.crosstab([df.C_Gender__c, df.C_Ethnic_background__c], df.RT_RecordType__c, margins=True, colnames=[
            'Student Type'], rownames=['Gender', 'Ethnicity'], margins_name='Total')

Student Type              College Student  High School Student  Total
Gender Ethnicity                                                     
Female African-American               140                   78    218
       Asian-American                   0                    8      8
       Decline to State                 0                    2      2
       Latino / Chicano                11                   10     21
       Multiracial                      1                    3      4
       Native American                  0                    1      1
       Other                            0                    4      4
       White / Caucasian                0                    1      1
Male   African-American                51                   27     78
       Asian-American                   1                    4      5
       Latino / Chicano                 2                    4      6
       Multiracial                      2                    1      3
       Other                            0                    1      1
       White / Caucasian                0                    3      3
Other  Asian-American                   0                    1      1
Total                                 208                  148    356

#### Count of unduplicated participants by sex and Employment Status


In [54]:
pd.crosstab([df.C_Gender__c, df.C_Employment_Status__c], df.RT_RecordType__c, margins=True, colnames=[
            'Student Type'], rownames=['Gender', 'Employment Status'], margins_name='Total')

Student Type                     College Student  High School Student  Total
Gender Employment Status                                                    
Female Employed part time                      3                    0      3
       No information available              149                  107    256
Male   No information available               56                   40     96
Other  No information available                0                    1      1
Total                                        208                  148    356

#### *Count of unduplicated participants by household income in the following categories: Total Number of Persons less than or equal to 30% Area Median Income (extremely low income), Total Number of Persons over 30% not greater than 50% Area Median Income (low income), Total Number of Persons over 50% not greater than 80% Area Median Income (moderate income), Total Number of Persons over 80% Area Median Income (non-LMI), No Income, Unknown.

In [10]:
pd.crosstab([df['Income Bucket']], df.RT_RecordType__c, margins=True, colnames=[
            'Student Type'], rownames=['Income Bucket'], margins_name='Total')

Student Type                                College Student  \
Income Bucket                                                 
Between 30% of 50% of Median Income                      25   
Between 51% and 80% of Median Income                     62   
Greater than 80% of Median Income                        87   
Less than or equal to 30% of Median Income               34   
Total                                                   208   

Student Type                                High School Student  Total  
Income Bucket                                                           
Between 30% of 50% of Median Income                          18     43  
Between 51% and 80% of Median Income                         55    117  
Greater than 80% of Median Income                            56    143  
Less than or equal to 30% of Median Income                   19     53  
Total                                                       148    356

#### Gender / Parish Breakdown



In [11]:
pd.crosstab([df.C_Gender__c, df.Parish.fillna('Missing')], df.RT_RecordType__c, margins=True, colnames=[
            'Student Type'], rownames=['Gender', 'Parish'], margins_name='Total')

Student Type                    College Student  High School Student  Total
Gender Parish                                                              
Female East Baton Rouge Parish                1                    0      1
       Jefferson Parish                      17                    8     25
       Missing                                5                    0      5
       Orleans Parish                       125                   97    222
       Plaquemines Parish                     0                    1      1
       St. Bernard Parish                     2                    0      2
       St. Tammany Parish                     2                    1      3
Male   Harris County                          1                    0      1
       Jefferson Parish                       8                    2     10
       Orleans Parish                        46                   38     84
       St. Charles Parish                     1                    0      1
Other  Orleans Parish                         0                    1      1
Total                                       208                  148    356

## High School Questions

#### *# and % of high school students promoted to the next grade from the 2019-2020 academic year

This should be 100% or 148 (all HS students)


In [20]:
df_2021 = df[df.C_HIGH_SCHOOL_GRADUATING_CLASS__c == "2022"]

#### *% and # of class of 2021 seniors who graduated high school on time


This is 100% or 59 students (from the DDT)

#### *% and # of Class of 2022 NOLA seniors who have submitted, completed, or are in the “review” phase with FAFSA to date

In [21]:
def create_count_percent_cross_tab(df, row_column, column_column, row_name, subset_name = 'High School Student'):
    cross_1 = pd.crosstab(df[row_column].fillna('No Data'), df[column_column], dropna=False, colnames=[
            subset_name], rownames=[row_name], margins=True)
    
    cross_2 = pd.crosstab(df[row_column].fillna('No Data'), df[column_column], dropna=False, colnames=[
            subset_name], rownames=[row_name], normalize=True, margins=True)
    
    cross_1['Percent'] = cross_2[subset_name].values
    cross_1.rename(columns={subset_name: "Count"}, inplace=True)
    cross_1.drop('All', inplace=True, axis=1)
    return cross_1.style.format({'Percent': "{:.0%}"})


In [22]:
df_2021_us_citizen = df_2021[df_2021.C_Citizen_c__c == 'US Citizen']
df_2021_us_citizen_pr = df_2021[df_2021.C_Citizen_c__c != 'Other']

Not Including permanent residence.

In [23]:
create_count_percent_cross_tab(df_2021_us_citizen, 'C_FA_Req_FAFSA__c', 'RT_RecordType__c', 'FAFSA Status')

Including permanent residence.

In [24]:
create_count_percent_cross_tab(df_2021_us_citizen_pr, 'C_FA_Req_FAFSA__c', 'RT_RecordType__c', 'FAFSA Status')

#### % and # of our Class of 2022 NOLA high school seniors who qualify for TOPS eligibility.

It looks like this data isn't entered into the system yet.

In [28]:
create_count_percent_cross_tab(df_2021, 'C_Region_Specific_Funding_Eligibility__c', 'RT_RecordType__c', 'TOPS Status')

#### # and % of four-year college acceptances for the high school Class of 2022 during the reporting period.


In [29]:
df_2021['accepted_college'] = df.C_Four_Year_College_Acceptances__c > 0

In [30]:
create_count_percent_cross_tab(df_2021, 'accepted_college', 'RT_RecordType__c', 'Accepted into College')

#### # and % of NOLA high school students in the 2021-22 academic year with 3.0+ cumulative GPAs

The first table is based on the most recent GPA value we have on file for students. Which is mostly Spring 2020-21 data. 


The second table is based only Fall 2021-22 data, which isn't entered yet and won't be until the end of February.

In [35]:
df_hs = df[df.RT_RecordType__c == "High School Student"]


In [36]:
df_hs['above_3_gpa_most_recent'] = df.C_Most_Recent_GPA_Cumulative__c >= 3.0
df_hs['above_3_gpa_fall'] = df.AS_GPA_HS_cumulative__c >= 3.0

In [37]:
create_count_percent_cross_tab(df_hs, 'above_3_gpa_most_recent', 'RT_RecordType__c', 'Most Recent GPA >= 3.0')

In [38]:
create_count_percent_cross_tab(df_hs, 'above_3_gpa_fall', 'RT_RecordType__c', 'Spring 2020-21 GPA >= 3.0')

#### % of NOLA freshmen and sophomores required to attend MathBlast who completed the 3-5 weeks during this reporting period (summer 2021)

I found this in the DDT. It looks like 66 students were required to take Math Blast, and they all attended 100% of their sessions! 

In [51]:
# df_2022_2023 = df[(df.C_HIGH_SCHOOL_GRADUATING_CLASS__c == "2024") | (df.C_HIGH_SCHOOL_GRADUATING_CLASS__c == "2023")]

In [52]:
# df_2022_2023 = df_2022_2023[df_2022_2023.attendance_rate > 0]

In [50]:
# create_count_percent_cross_tab(df_2022_2023, 'above_80', 'RT_RecordType__c', 'Attended 80%+ Math Blast')

#### average # of four-year colleges to which Class of 2022 NOLA high school seniors applied during this reporting period



In [43]:
df_2021.C_Four_Year_College_Applications__c.mean().round(2)

4.59

#### average # of outside scholarships for which Class of 2022 NOLA high school seniors applied during this reporting period

In [44]:
df_2021.num_scholarships.mean().round(2)

1.0

#### # of avg. community service hours Class of 2022 NOLA high school seniors have completed throughout high school.

In [45]:
df_2021.C_Total_Community_Service_Hours_Completed__c.mean().round(2)

76.67

#### % of NOLA high school students who attend 80% or more of their scheduled College Track programming sessions during this reporting period

If we just use Fall 2021-22 data (which would be most stable) it would be 116 students (or 78% of students).

If you want to combine Summer 2020-21 and Fall 2021-22 data it would be 95% of students (or 149 students). Note that is pretty inflated by the good attendance in short sessions like Math Blast and Summer Bridge. I'd probably recomend not combining the dates, but it is your call! 




## College Questions

#### % of rising college freshmen who completed Summer Bridge to prepare for matriculation during this reporting period

I also found this in the DDT. It looks like 56 studens attended Summer Bridge and 80% (45) of them attended above 80% of sessions. Overall they had 59 seniors, so 95% of students participate.


In [46]:
# df_2020 = df[(df.C_HIGH_SCHOOL_GRADUATING_CLASS__c == "2021")]

In [49]:
# create_count_percent_cross_tab(df_2020, 'above_80', 'RT_RecordType__c', 'Attended 80%+ Summer Bridge',subset_name='High School Student')

## Questions with No Data Related Answer

#### *% of seniors who successfully complete recommended dual enrollment classes
I'd recomend asking sites as you indicated you are doing. 


**Question:**

* The number of households that fit within the following "Client Type of Household" categories: Couple With Children/Dependents, Couple With No Children/Dependents, Female Householder With Children/Dependents - No Spouse/Partner, Male Householder with Children/Dependents - No Spouse/Partner, Single Household, Other, Unknown
--from our last report, I understand most if not all of our student households fall into the "unknown" category. Please confirm that this is still the case.

**Answer: All of our students would fall into the unknown category**

In [1]:
%%html

<script>
$(document).ready(function(){
    window.code_toggle = function() {
        (window.code_shown) ? $('div.input').hide(250) : $('div.input').show(250);
        window.code_shown = !window.code_shown
    }
    if($('body.nbviewer').length) {
        $('<li><a href="javascript:window.code_toggle()" title="Show/Hide Code"><span class="fa fa-code fa-2x menu-icon"></span><span class="menu-text">Show/Hide Code</span></a></li>').appendTo('.navbar-right');
        window.code_shown=false;
        $('div.input').hide();
    }
});
</script>


<style>

div.prompt {display:none}


h1, .h1 {
    font-size: 33px;
    font-family: "Trebuchet MS";
    font-size: 2.5em !important;
    color: #2a7bbd;
}

h2, .h2 {
    font-size: 10px;
    font-family: "Trebuchet MS";
    color: #2a7bbd; 
    
}


h3, .h3 {
    font-size: 10px;
    font-family: "Trebuchet MS";
    color: #5d6063; 
    
}

.rendered_html table {

    font-size: 14px;
}

.output_png {
  display: flex;
  justify-content: center;
}

.cell {
    padding: 0px;
}


</style>